src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1749,32 +1749,37 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
         OneOf(
             # Columns and comment syntax:
             Sequence(
-                Bracketed(
-                    Delimited(
-                        OneOf(
-                            Sequence(
-                                Ref("ColumnReferenceSegment"),
-                                Ref("DatatypeSegment"),
-                                AnyNumberOf(
-                                    # A single COLLATE segment can come before or after
-                                    # constraint segments
-                                    OneOf(
-                                        Ref("ColumnConstraintSegment"),
-                                        Sequence(
-                                            "COLLATE",
-                                            Ref("CollationReferenceSegment"),
+                OneOf(
+                    Bracketed(
+                        Delimited(
```

Comment:
If we stick an `optional=True` here, we can remove the outer `OneOf`, and the empty `Bracketed()` below

Reply:
Like this? The tests don't parse in that case: https://github.com/sqlfluff/sqlfluff/pull/5100/commits/397cae6545759f15324e6eae9974eb7c757a29ca